dissect/target/plugins/os/unix/bsd/osx/_os.py

```diff
@@ -30,35 +30,49 @@ def create(cls, target: Target, sysvol: Filesystem) -> MacPlugin:
 
     @export(property=True)
     def hostname(self) -> Optional[str]:
-        for path in ["/Library/Preferences/SystemConfiguration/preferences.plist"]:
-            try:
-                preferencesPlist = self.target.fs.open(path).read().rstrip()
-                preferences = plistlib.loads(preferencesPlist)
-                return preferences["System"]["System"]["ComputerName"]
+        try:
+            preferencesPlist = self.target.fs.path(self.SYSTEM).read_bytes()
+            preferences = plistlib.loads(preferencesPlist)
+            return preferences["System"]["System"]["ComputerName"]
 
-            except FileNotFoundError:
-                pass
+        except FileNotFoundError:
+            pass
 
     @export(property=True)
     def ips(self) -> Optional[list[str]]:
         raise NotImplementedError
 
     @export(property=True)
     def version(self) -> Optional[str]:
-        for path in ["/System/Library/CoreServices/SystemVersion.plist"]:
-            try:
-                systemVersionPlist = self.target.fs.open(path).read().rstrip()
-                systemVersion = plistlib.loads(systemVersionPlist)
-                productName = systemVersion["ProductName"]
-                productUserVisibleVersion = systemVersion["ProductUserVisibleVersion"]
-                productBuildVersion = systemVersion["ProductBuildVersion"]
-                return f"{productName} {productUserVisibleVersion} ({productBuildVersion})"
-            except FileNotFoundError:
-                pass
+        try:
+            systemVersionPlist = self.target.fs.path(self.VERSION).read_bytes()
+            systemVersion = plistlib.loads(systemVersionPlist)
+            productName = systemVersion["ProductName"]
+            productUserVisibleVersion = systemVersion["ProductUserVisibleVersion"]
+            productBuildVersion = systemVersion["ProductBuildVersion"]
+            return f"{productName} {productUserVisibleVersion} ({productBuildVersion})"
+        except FileNotFoundError:
+            pass
 
     @export(record=UnixUserRecord)
     def users(self) -> Iterator[UnixUserRecord]:
-        raise NotImplementedError
+        for path in self.target.fs.path("/var/db/dslocal/nodes/Default/users/").glob("*.plist"):
+            user = plistlib.loads(path.read_bytes())
```

Comment:
```suggestion
            user = plistlib.load(path.open())
```
This doesn't work? Also applies for all the other `plistlib.loads` instances if it does work :)

Reply:
Both options work